scaaml/capture/input_generators/attack_point_iterator.py

```diff
@@ -279,3 +281,66 @@ def get_generated_keys(self) -> List[str]:
         for operand in self._operands:
             generated_keys += operand.get_generated_keys()
         return generated_keys
+
+
+class AttackPointIteratorCartesianProduct(AttackPointIterator):
+    """
+    Attack point iterator cartesian product class. This class takes any amount
+    of operands and combines them just like a cartesian product would.
+    """
+
+    def __init__(self, operation: str, operands: List[Dict[str, Any]]) -> None:
+        """Initialize the cartesian product iterator.
+          
+          Args:
+            operation (str): The operation of the iterator
+                represents what the iterator does and what 
+                has to be in the config file. This is only used once to
+                double check if the operation is the correct one.
+                
+            operands (List[Dict[str, Any]]): The operands are any number of
+                iterator configs that will be combined."""
+        assert "cartesian_product" == operation
+        if not operands:
+            raise ValueError
+        elif len(operands) > 1:
+            self._operands = [
+                build_attack_points_iterator(operands[0]),
+                build_attack_points_iterator({
+                    "operation": "cartesian_product",
+                    "operands": operands[1:]
+                })
+            ]
+        else:
+            self._operands = [build_attack_points_iterator(operands[0])]
+        operand_lengths = [operand._len for operand in self._operands]
+        if any(length < 0 for length in operand_lengths):
+            raise LengthIsInfiniteException
+        elif not self._operands or any(
+                length == 0 for length in operand_lengths):
+            self._len = 0
+        else:
+            self._len = math.prod(operand_lengths)
+
+    def __iter__(self) -> AttackPointIteratorT:
+        if len(self._operands) == 2:
+            return iter(
+                self._merge_dictionaries(value_one, value_two)
```

Comment:
The `_merge_dictionaries` is pretty much a copy-paste.
```suggestion
                {**value_one, **value_two}
```